### PR TITLE
Upgrade intel-pf-bb-config to 23.11

### PR DESCRIPTION
**Pull request by vinceaperri**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 
Upgrading package to latest version.

###### Change Log  
- Upgrade intel-pf-bb-config to 23.11
- Replace build.sh script with actual commands since git isn't available in build stage anymore
- Replace git command at the start of the build steps with hardcoded version tag, replacing and simplifying code

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Pipeline build id: [503062](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=503062&view=results)

### CODE REVIEW

File: SPECS/intel-pf-bb-config/intel-pf-bb-config.spec [hunk 1]
```diff
@@ -19,7 +22,8 @@ accessing their configuration space and setting parameters via MMIO.
 %autosetup -n pf-bb-config-%{version}
 
 %build
-./build.sh
+sed -i "s/#VERSION_STRING#/v%{version}-0-gf9206ad/g" config_app.c
```
Comment (romoh) [inline SPECS/intel-pf-bb-config/intel-pf-bb-config.spec:25]: why do we need this? maybe worth adding a comment

**Comment by amritakohli** [comment]
cgmanifest should be updated

**Review by amritakohli — CHANGES_REQUESTED** [review]
cgmanifest should be updated
nit: clean up description to remove optional parts that aren't filled out

File: SPECS/intel-pf-bb-config/intel-pf-bb-config.spec (hunk 1 above)
Comment (vinceaperri) [reply SPECS/intel-pf-bb-config/intel-pf-bb-config.spec:25]: It's directly from the build.sh script

**Review by romoh — APPROVED** [review]

**Review by amritakohli — APPROVED** [review]

**Pull request merged**